src/lib/storage/snapshots_finish.rb

```diff
@@ -0,0 +1,48 @@
+require "yast2/fs_snapshot"
+require "installation/finish_client"
+
+module Storage
+  class SnapshotsFinish < ::Installation::FinishClient
+    include Yast::I18n
+
+    def initialize
+      textdomain "storage"
+
+      Yast.import "Mode"
+      Yast.import "StorageSnapper"
+      Yast.include self, "installation/misc.rb"
+    end
+
+    # Write configuration
+    #
+    # It creates a snapshot when no second stage is required and
+    # Snapper is configured.
+    #
+    # @return [TrueClass,FalseClass] True if snapshot was created;
+    #                                otherwise it returns false.
+    def write
+      if !second_stage_required? && StorageSnapper.configure_snapper?
+        log.info("Creating root filesystem snapshot")
+        action = Mode.update ? "upgrade" : "installation"
+        create_snapshot(:single, "after #{action}")
```

Comment:
I've though about implementing that as part of this PBI https://trello.com/c/E8LbHch1/143-5-12-sp1-m-317900-create-snapshot-before-starting-upgrade. I guess I need the number of the `pre` snapshot to pair with it.